### PR TITLE
Fix/path concat

### PR DESCRIPTION
**Pull request by scinos**
#### Changes proposed in this Pull Request

* Enables eslint rule [no-path-concat](https://eslint.org/docs/rules/no-path-concat)
* Fixes violations of said rule
* Add custom .eslintrc file to `packages/wpcom.js` to allow server code in server example.

#### Testing instructions
N/A

### CODE REVIEW

File: packages/wpcom-proxy-request/examples/webpack.config.js [hunk 1]
```diff
@@ -1,12 +1,14 @@
+const path = require( 'path' );
+
 module.exports = {
-	entry: __dirname + '/../build/index.js',
+	entry: path.join( __dirname, '/../build/index.js' ),
```
Comment (jsnajdr) [inline packages/wpcom-proxy-request/examples/webpack.config.js:4]: The leading slashes are not needed and could suggest that the path is absolute. Is this an application of the rule's autofix, or did you make the changes manually?

File: packages/wpcom.js/examples/server/index.js [hunk 2]
```diff
@@ -2,41 +2,42 @@
...
 
 	// get site info
 	site.get( function ( err, info ) {
 		if ( err ) return console.log( err );
 
 		// get lastest posts
-		site.postsList( { number: 10 }, function ( err, posts ) {
-			if ( err ) return console.log( err );
+		site.postsList( { number: 10 }, function ( error, posts ) {
+			if ( error ) return console.log( error );
```
Comment (jsnajdr) [inline packages/wpcom.js/examples/server/index.js:40]: This `err` scope problem could also be fixed by promisifying the code. Creating an example of the `wpcom` promise API 🙂

**Review by jsnajdr — APPROVED** [review]
:+1:

File: packages/wpcom-proxy-request/examples/webpack.config.js (hunk 1 above)
Comment (scinos) [reply packages/wpcom-proxy-request/examples/webpack.config.js:4]: I wrote a regexp to autofix it (the rule doesn't provide any). I can change it to remove the initial `/`

**Pull request merged**